feat: petting zoo style env

LovelyBuggies opened the pull request.

LovelyBuggies committed:
refactor: rl example

LovelyBuggies committed:
Merge branch 'main' into rllib-test

LovelyBuggies committed:
refactor: change name

LovelyBuggies committed:
Merge branch 'main' into rllib-test

LovelyBuggies committed:
feat: step func

LovelyBuggies committed:
Merge branch 'main' into rllib-test

wyunhao committed:
param and plot rewards

LovelyBuggies committed:
fix:

LovelyBuggies committed:
feat: finish spaces

LovelyBuggies committed:
fix: -1000 for negative battery

LovelyBuggies committed:
Update fmp-numerical.py

LovelyBuggies committed:
Create rllib-test-fmp.py

LovelyBuggies committed:
fix: use inner func

LovelyBuggies committed:
FEAT: modify observation space and connect to SB3

LovelyBuggies committed:
refactor: file struct

LovelyBuggies committed:
feat: add ppo test result

wyunhao committed:
add dqn

wyunhao committed:
add dqn and jumbo env

qqqube committed:
dqn tensorboard logs

LovelyBuggies committed:
feat: add ppo tensorboard

LovelyBuggies committed:
feat:

wyunhao committed:
update latest model result

wyunhao committed:
update jumbo ppo data and model

wyunhao committed:
change setting to pettingzoo customized env

wyunhao committed:
fix self.observations

LovelyBuggies committed:
Update sb3.py

LovelyBuggies committed:
Merge branch 'petting-zoo-style-env' of https://github.com/LovelyBuggies/sumo-gym into petting-zoo-style-env

LovelyBuggies committed:
Update sb3.py

wyunhao committed:
remove redundant ob def

wyunhao committed:
fix action space conversion

wyunhao committed:
fix raw structure of step function

wyunhao committed:
delete inner_reset

wyunhao committed:
fix reward update

LovelyBuggies committed:
Update fmp.py

LovelyBuggies committed:
Update fmp.py

wyunhao committed:
update action space len logic

LovelyBuggies committed:
Update fmp.py

LovelyBuggies committed:
Update fmp.py

LovelyBuggies committed:
Update fmp.py

LovelyBuggies committed:
fmp&fmp-num

LovelyBuggies committed:
reward problem

LovelyBuggies committed:
Update fmp.py

wyunhao committed:
small fix

wyunhao committed:
fix prev_is_loading

LovelyBuggies committed:
Update fmp.py

wyunhao committed:
fix obs

wyunhao committed:
fix ob space len

wyunhao committed:
fix action space

wyunhao committed:
remove print

wyunhao committed:
ghost idle action

LovelyBuggies committed:
Update fmp-numerical.py

wyunhao committed:
two vehicle for numerical test

wyunhao committed:
fix multi-vehicle

wyunhao committed:
three vehicle for numerical

LovelyBuggies committed:
update

LovelyBuggies committed:
fix: sb3

LovelyBuggies committed:
finish tutorial

LovelyBuggies committed:
Delete test_env.py

LovelyBuggies committed:
Update fmp-numerical.py

LovelyBuggies committed:
tutorial

LovelyBuggies committed:
Update fmp-numerical.py

LovelyBuggies committed:
add range for tutorial

LovelyBuggies committed:
Update sb3.py

LovelyBuggies committed:
feat: onboard sb3 a2c

LovelyBuggies committed:
Update fmp.py

wyunhao committed:
add inner step for agent with no action taken

wyunhao committed:
update previous info

wyunhao committed:
inner step condition fix

wyunhao committed:
agent info

LovelyBuggies committed:
Update fmp-numerical.py

wyunhao committed:
update reward update

wyunhao committed:
reduce reward if no move performed

wyunhao committed:
small fix

LovelyBuggies committed:
Update fmp.py

LovelyBuggies committed:
Update fmp.py

wyunhao committed:
log

wyunhao committed:
fix the agent selector issue and responded update

wyunhao committed:
log

wyunhao committed:
add reward to tensorboard, though all zero

wyunhao committed:
remove duplicate reward calculation function

wyunhao committed:
reward update

wyunhao committed:
fix reset

wyunhao committed:
fix reward zero update issue

wyunhao committed:
remove incomplete tensorboard log

wyunhao committed:
new reward curve

LovelyBuggies committed:
Update fmp.py

wyunhao commented:
Work in this PR:

- [x] remove duplication and change to vector-like (Discrete, list, ...) data structure to speed up
- [x] ~~self.dones termination can only be completed at the end of the cycle due to the restriction of parallel environment~~ now comes to: all variables are updated in each step of an agent (we are going to create training parts by ourselves)
- [x] charging should not happen when at full capacity
- [x] shrink state space of battery level and tune it to fully discrete (https://github.com/LovelyBuggies/sumo-gym/pull/55#issuecomment-1013979638)
- [x] when in low-battery, only done for one agent and continue the process until all responded
- [x] need to change the self.responded data update, rewards should not be added to vehicle even if the demand is already responded by other vehicles. 
- [x] better to keep a queue for CS, cannot accommodate too many EVs -- if an EV stepped into a fully-loaded CS, it should wait (still have a reward discount) and would not be charged until one moved. 

---

TBD in next PR:

- [ ] model trained should be adapted to our own algorithm
- [ ] get the decreasing loss curve and increasing cumulative reward curve (in the ideal case)
- [ ] reward curves for all agents with different electricity-cost settings ("stronger works more")
- [ ] make some running examples after training based on the policy ("stronger works more" and small-capacity EV not responded to long-distance response)

wyunhao committed:
fix termination for parallel env

LovelyBuggies committed:
Update fmp.py

LovelyBuggies committed:
Merge branch 'petting-zoo-style-env' of https://github.com/LovelyBuggies/sumo-gym into petting-zoo-style-env

LovelyBuggies committed:
Update fmp.py

LovelyBuggies committed:
style:

LovelyBuggies committed:
fix: reset states to departure

LovelyBuggies committed:
fix: recalculate rewards

LovelyBuggies committed:
feat: change data structure (#54)

* feat: renovate

* fix: move return obs[4]

* Update fmp.py

* fix reward

* rm nearest_cs_with_dist and FMPState from fmp_utils

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* feat: add cumulative

* Create test.py

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Update test.py

* add test

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Update requirements.txt

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

LovelyBuggies committed:
feat: shrink the state space (#55)

* merge two dimension in states to one

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* feat: add n_demands

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* fix: tests

* Update test_fmp.py

* feat: update EV settings

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* np to list

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Update fmp-numerical.py

* Update fmp.py

* feat: add bat level

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

wyunhao committed:
Merge branch 'main' into petting-zoo-style-env

LovelyBuggies committed:
feat: fix observation

LovelyBuggies committed:
feat: now even "no move", still have a time cost

wyunhao committed:
fix the rewards adding issue

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

wyunhao committed:
fix the low battery issue

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

LovelyBuggies committed:
Update fmp.py

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

wyunhao committed:
implement the k-means util func

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

LovelyBuggies committed:
Update requirements.txt

LovelyBuggies committed:
Update fmp_utils.py

LovelyBuggies committed:
refactor: remove sb3

LovelyBuggies committed:
refactor: remove deprecated graph support and sb3

LovelyBuggies committed:
feat: apply cs slots and move self.responded inside (#57)

* feat: apply cs slots and move self.responded inside

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>
